File: app/src/main/java/com/marked/pixsee/signup/DialogRegistration.java
package com.marked.pixsee.signup;


import android.app.ProgressDialog;
import android.content.Context;
import android.widget.Toast;

import com.marked.pixsee.networking.HTTPStatusCodes;

/**
 * Created by Tudor Pop on 23-Jan-16.
 */
public class DialogRegistration  {
	private ProgressDialog progressDialog ;
	private Context context ;

	public DialogRegistration(ProgressDialog progressDialog, Context context) {
		this.progressDialog = progressDialog;
		this.context = context;
	}

	public void onError(int errorStatusCode) {
		switch (errorStatusCode){
			case HTTPStatusCodes.REQUEST_CONFLICT:
				Toast.makeText(context,"You already have an account",Toast.LENGTH_SHORT).show();
				break;
			case HTTPStatusCodes.REQUEST_TIMEOUT:
				Toast.makeText(context,"Timeout error",Toast.LENGTH_SHORT).show();
				break;
			case HTTPStatusCodes.UNPROCESSABLE_ENTITY:
				Toast.makeText(context,"Incorrect password",Toast.LENGTH_SHORT).show();
				break;
			case HTTPStatusCodes.NOT_FOUND:
				Toast.makeText(context,"We are sorry, but we did not found you",Toast.LENGTH_SHORT).show();
				break;

		}
	}

}

File: app/src/main/java/com/marked/vifo/delegate/DialogRegistration.kt
package com.marked.vifo.delegate

import android.app.ProgressDialog
import android.content.Context
import com.marked.vifo.extra.HTTPStatusCodes
import com.marked.vifo.gcm.RegistrationListener
import org.jetbrains.anko.toast

/**
 * Created by Tudor Pop on 23-Jan-16.
 */
class DialogRegistration(context: Context, progressDialog: ProgressDialog) : RegistrationListener {
	private val progressDialog = progressDialog
	private val context = context

	override fun onDismiss() {
		progressDialog.dismiss()
	}

	override fun onError(errorStatusCode: Int) {
		when (errorStatusCode) {
			HTTPStatusCodes.REQUEST_CONFLICT -> context.toast("You already have an account")
			HTTPStatusCodes.REQUEST_TIMEOUT -> context.toast("Timeout error")
			HTTPStatusCodes.UNPROCESSABLE_ENTITY -> context.toast("Incorrect password")
			HTTPStatusCodes.NOT_FOUND -> context.toast("We are sorry, but we did not found you")
		}
	}
}